ft-main/io/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+#![no_std]
+use ft_logic_io::Action;
+use gstd::{prelude::*, ActorId};
+use primitive_types::H256;
+
+#[derive(Encode, Decode, TypeInfo, Debug)]
+pub enum FTokenAction {
+    Message {
+        transaction_id: u64,
+        payload: Action,
+    },
+    UpdateLogicContract {
+        ft_logic_code_hash: H256,
+        storage_code_hash: H256,
+    },
+    MigrateStorageAddresses,
```

Comment:
What's this one for?

Reply:
For saving storage addresses when updating the logic contract. That function isn't implemented yet